packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -154,7 +166,13 @@ export let visitor = {
         for (let candidate of (state.candidates: Array)) {
           candidate.replaceWith(argsId);
           if (candidate.parentPath.isMemberExpression()) {
-            optimiseMemberExpression(candidate.parent, state.offset);
+            if (t.isReturnStatement(candidate.parentPath.parent)
```

Comment:
There's way too much nesting here, can you abstract it out into a function or something?